hvpy/helpers.py

```diff
@@ -0,0 +1,123 @@
+    scaleY: Optional[float] = None,
+    movieLength: Optional[float] = None,
+    watermark: Optional[bool] = True,
+    width: Optional[str] = None,
+    height: Optional[str] = None,
+    x0: Optional[str] = None,
+    y0: Optional[str] = None,
+    x1: Optional[str] = None,
+    y1: Optional[str] = None,
+    x2: Optional[str] = None,
+    y2: Optional[str] = None,
+    size: Optional[int] = None,
+    movieIcons: Optional[int] = None,
+    followViewport: Optional[int] = None,
+    reqObservationDate: Optional[datetime] = None,
+    overwrite: bool = False,
+    filename: Union[str, Path] = None,
+    hq: bool = False,
+    timeout: float = 5,
+) -> Path:
+    """
+    Automatically creates a movie using `queueMovie`, `getMovieStatus` and
+    `downloadMovie` functions.
+
+    Parameters
+    ----------
+    {Insert}
+    overwrite
+        Whether to overwrite the file if it already exists.
+        Default is `False`.
+    filename
+        The path to save the file to.
+        Optional, will default to ``f"{starttime}_{endtime}.{format}"``.
+    hq
+        Download a higher-quality movie file (valid for "mp4" movies only, ignored otherwise).
+        Default is `False`, optional.
+    timeout
+        The timeout in minutes to wait for the movie to be created.
+        Default is 5 minutes.
+
+    Examples
+    --------
+    >>> from hvpy import createMovie, DataSource, create_events, create_layers
+    >>> from datetime import datetime, timedelta
+    >>> createMovie(
+    ...     startTime=datetime.today() - timedelta(days=15, minutes=5),
+    ...     endTime=datetime.today() - timedelta(days=15),
+    ...     layers=create_layers([(DataSource.AIA_171, 100)]),
+    ...     events=create_events(["AR"]),
+    ...     eventsLabels=True,
+    ...     imageScale=1,
+    ...     filename="my_movie",
+    ... )
+    ...Path('my_movie.mp4')
+    """
+    input_params = locals()
+    # These are used later on but we want to avoid passing
+    # them into queueMovie.
+    input_params.pop("overwrite")
+    input_params.pop("filename")
+    input_params.pop("hq")
+    input_params.pop("timeout")
+    res = queueMovie(**input_params)
+    if res.get("error"):
+        raise RuntimeError(res["error"])
+    timeout_counter = time.time() + 60 * timeout  # Default 5 minutes
+    while True:
+        status = getMovieStatus(
+            id=res["id"],
+            format=format,
+            token=res["token"],
+        )
+        if status["status"] in [0, 1]:
+            time.sleep(3)
+        if status["status"] == 2:
+            break
+        if time.time() > timeout_counter:
+            raise RuntimeError(f"Exceeded timeout of {timeout} minutes.")
+        if status["status"] == 3:
+            raise RuntimeError(status["error"])
+    binary_data = downloadMovie(
+        id=res["id"],
+        format=format,
+        hq=hq,
+    )
+    if filename is None:
+        filename = f"{res['id']}_{startTime.date()}_{endTime.date()}.{format}"
```

Comment:
I think that will be ok for now, but I think we need to consider a better way of creating filenames in future.

Reply:
Helioviewer.org's movie file names aren't much better
`2022_08_10_03_39_41_2022_08_10_14_19_41_AIA_304-hq.mp4` 

I think this is ok for now. If the user wants a clean filename, they can specify it. Having the ID here guarantees it will be unique, so that's a good call.

Reply:
That is a filename alright.